### ISSUE TITLE
Shortcuts for executables and tests

### ISSUE THREAD

**Issue by tfausak** [sol/hpack#161]
I think it would be nice to have shortcuts for defining executables and tests (and probably benchmarks too). Pretty much every package I define with executables and tests looks like this:

``` yaml
name: some-package
library: ...
executables:
  some-package: ...
tests:
  test: ...
```

That is, I have a single executable with the same name as the package and a single test suite with the name "test". I always wish it could be shortened to this:

``` yaml
name: some-package
library: ...
executable: ...
test: ...
```

The top level `executable: x` would be transformed into `executables: { some-package: x }`, and the `test` part would do the same. 

Is there any interest in this feature?

**Comment by sol** [comment]
Hi!  Sorry for the late reply.  I'm on the fence with this one.  I see how this is convenient but there are also costs / downsides:

 - code complexity
 - more complex user documentation
 - what happens if both `executable` and `executables` is specified?

That said, I see the biggest benefit for `executable`, as here we actually have duplicated information that we would like to DRY up (package name and executable name are the same, but we specify it twice).

For `tests` and `benchmarks` the benefit is less clear to me, as I see it we would mainly save one line in the package description.

Given this, I would be interested to see an implementation for `executable` first, before making a final call.

**Comment by tfausak** [comment]
Fixed by f405602e564ba890921f3c6fec01908b771b1541 and fda6b8be03a23395f59e7950c258bf8b328e0faa. See also #177 and #178.
